platform/ios/src/MGLMapView.mm

```diff
@@ -1556,18 +1556,20 @@ - (void)handleDoubleTapGesture:(UITapGestureRecognizer *)doubleTap
     if (doubleTap.state == UIGestureRecognizerStateEnded)
     {
         MGLMapCamera *oldCamera = self.camera;
-        
+
+        double newZoom = round(self.zoomLevel) + 1.0;
```

Comment:
Went a bit further on macOS in 4ed80f9 and rounded all of the non-freeform zoom methods.